core/trino-spi/src/main/java/io/trino/spi/metrics/Metrics.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.Mergeable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class Metrics
+        implements Mergeable<Metrics>
+{
+    public static final Metrics EMPTY = new Metrics(Map.of());
+
+    private final Map<String, Metric> metrics;
+
+    @JsonCreator
+    public Metrics(Map<String, Metric> metrics)
+    {
+        this.metrics = requireNonNull(metrics, "metrics is null");
```

Comment:
This class is holding on to the map provided by the caller. That can cause potential problems down the road if the map is modified (especially in a non-thread-safe manner). Do `Map.copyOf(...)`

Reply:
Thanks, fixed.